### DISCUSSION TITLE
Include the seal as a png file under the tiff file.

### DISCUSSION THREAD

**Discussion by gsgalezowski** [ImageMagick/ImageMagick#6419] [Help]
### ImageMagick version

7.0.9-Q16

### Operating system, version and so on

Windows 10

### Description

I need to combine two files into one (png and tiff uncompressed). The tiff file should be above the png file. Merging involves one png file and multiple tiff files (uncompressed). Each tiff file needs to be merged only with that particular png file.
I prepared a script for mass merging files:
for %%z in (*.png) do (
magick "%%z " test.png
-resize %%
-gravity north -geometry +0+0 -append *.png %%test.tiff"
)
I receive tiff files but with a huge size. 
The tiff file should be uncompressed and in 24-bit color palette.
After executing the convert +matte -type truecolor -depth file_x file_y command. I get a tiff file with a 24 bit palette but only a few megabytes smaller.
I don't know how to do the merge process without increasing the file size?

**Comment by snibgo** [comment]
I don't understand what you are trying to do.

Your command loops through *.png, executing magick for each one. But every magick command reads *.png, so it reads all the png files, and appends them. I don't think that is what you want.

Your output name `%%test.tiff` is weird.

Perhaps you want to write a png file and a tiff file. If so, then your command might end like this:
```
... -write somefile.png otherfile.tiff
```

**Comment by gsgalezowski** [comment]
I combine the two files into one and a sort of watermark is created. The files must not overlap. Therefore, the watermark option cannot be used. The size of the files when merging must be adjusted in the resulting file so that the whole is readable. The output must be tiff without compression and in 24 bit color palette. The resulting file must not be larger than the files from which it was assembled.
In short: two files (png - watermark + uncompressed tiff) are vertically merged into one uncompressed tiff file. The tiff files are scanned A4 pages and the png file is a small scan of the stamp. Therefore, after mass merging, the whole thing must be readable.

**Comment by fmw42** [comment]
Please post your input files and an example of what you want for the output.

**Comment by gsgalezowski** [comment]
![folder](https://github.com/ImageMagick/ImageMagick/assets/1382700/6aadd45b-a563-4c1e-b2c9-e72868374278)
![file](https://github.com/ImageMagick/ImageMagick/assets/1382700/9826c81a-9fae-4f92-a856-7710beff9133)

**Comment by snibgo** [comment]
The magick command would look like this (Windows BAT syntax):
```
magick ^
  1.tiff ^
  stemple.png ^
  -append ^
  out.png
```
If you need to resize either input image, I suggest using parentheses, like this:
```
magick ^
  ( 1.tiff -resize 123x456 ) ^
  ( stemple.png -resize 123x34 ) ^
  -append ^
  out.png
```
If that command is inside a "`for (...)`" loop, you will need to escape the parentheses like this:
```
magick ^
  ^( 1.tiff -resize 123x456 ^) ^
  ^( stemple.png -resize 123x34 ^) ^
  -append ^
  out.png
```
When that command does what you want, you can put it inside a "`for (...)`" loop, using variable names instead of filenames.

**Comment by gsgalezowski** [comment]
The tiff file is displayed as a negative. It does not retain the colors of
the original. In addition, the output file should have a 24-bit color
palette when I choose tiff

pon., 26 cze 2023 o 15:18 snibgo ***@***.***> napisał(a):

> The magick command would look like this (Windows BAT syntax):
>
> magick ^
>   1.tiff ^
>   stemple.png ^
>   -append ^
>   out.png
>
> If you need to resize either input image, I suggest using parentheses,
> like this:
>
> magick ^
>   ( 1.tiff -resize 123x456 ) ^
>   ( stemple.png -resize 123x34 ) ^
>   -append ^
>   out.png
>
> If that command is inside a "for (...)" loop, you will need to escape the
> parentheses like this:
>
> magick ^
>   ^( 1.tiff -resize 123x456 ^) ^
>   ^( stemple.png -resize 123x34 ^) ^
>   -append ^
>   out.png
>
> When that command does what you want, you can put it inside a "for (...)"
> loop, using variable names instead of filenames.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ImageMagick/ImageMagick/discussions/6419#discussioncomment-6282189>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKRSLEHEFUGXS72ZWINAELXNGDZ3ANCNFSM6AAAAAAZOQWJHY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


--

**Comment by snibgo** [comment]
> The tiff file is displayed as a negative.

So, show a sample tiff file. Otherwise we have to guess.

I guess that the tiff file is encoded as CMYK, and the png file is encoded as sRGB. If so, I suggest you ensure they are both sRGB before appending them.
